src/server/ocpp/validation/OCPPValidation.ts

```diff
@@ -131,9 +134,10 @@ export default class OCPPValidation extends SchemaValidator {
         action: ServerAction.OCPP_METER_VALUES,
         message: `Connector ID '${meterValues.connectorId}' not found for Transaction ID '${meterValues.transactionId}'`
       });
+      return;
     }
     // Transaction ID is provided on Connector
-    if (foundConnector.currentTransactionID > 0) {
+    if (foundConnector?.currentTransactionID > 0) {
       // Check if provided in Meter Values
       meterValues.transactionId = Utils.convertToInt(meterValues.transactionId);
```

Comment:
Same as above, no need to convert it.

Reply:
Yes

Reply:
You can also change the code to not check the '0' with the Utils.isUndef..()

Reply:
in the if statement right ?

Reply:
> in the if statement right ?
Yes.
Test it with Posman.

Reply:
the case where the meterValue.transactionID is 0 is not check by utils.isNullorUndef, so if our value is 0, it will be not be true but we wait "true"

Reply:
> the case where the meterValue.transactionID is 0 is not check by utils.isNullorUndef, so if our value is 0, it will be not be true but we wait "true"

If meterValue.transactionID is filtered by the schema, it should be undefined when checked.
Test with Postman!